### DISCUSSION TITLE
Authentication Timeout in nats

### DISCUSSION THREAD

**Discussion by optiproplus** [amidaware/tacticalrmm#1763] [Q&A] [answered]
**Server Info (please complete the following information):**
 - OS: Ubuntu 22.04.3
 - Browser: firefox
 - RMM Version (as shown in top left of web UI): v0.17.5

**Installation Method:**
  - [x] Standard
  - [ ] Docker

**Describe the bug**:

1. this is fresh install to the new vps
2. client installs successfully, however i'm unable to control it
3. i can see constant timeouts in nats while in debug mode, here is http request trace between nginx and nat-server

T 127.0.0.1:55290 -> 127.0.0.1:9235 [AP] #147045
  GET /natsws HTTP/1.1..Host: api.xxx.yyy..Upgrade: websocket..Connection: upgrade..X-Forwarded-Host: api.xx
  x.yyy:443..X-Forwarded-For: a.b.c.d, e.f.g.h..X-Forwarded-Proto: https..X-Real-IP: a.b.c
  .d..User-Agent: Go-http-client/1.1..Sec-WebSocket-Key: o+oHYyk8+8vjllySpL45Cg==..Sec-WebSocket-Version: 13..Sec-Web
  socket-Extensions: permessage-deflate; server_no_context_takeover; client_no_context_takeover....                    
##
T 127.0.0.1:9235 -> 127.0.0.1:55290 [AP] #147047
  HTTP/1.1 101 Switching Protocols..Upgrade: websocket..Connection: Upgrade..Sec-WebSocket-Accept: X8/o3ZdLtmvpkARZM9NP
  IGbTaJc=....                                                                                                         
##
T 127.0.0.1:9235 -> 127.0.0.1:55290 [AP] #147049
  .~..INFO {"server_id":"NBC4ULHQ7VNMUFKYFPCUKB6CBYRSOKIQTPGDMJNS55MBAYULHAKB2OPZ","server_name":"NBC4ULHQ7VNMUFKYFPCUK
  B6CBYRSOKIQTPGDMJNS55MBAYULHAKB2OPZ","version":"2.10.10","proto":1,"git_commit":"983a1d2","go":"go1.21.6","host":"127
  .0.0.1","port":9235,"headers":true,"auth_required":true,"max_payload":67108864,"client_id":10345,"client_ip":"127.0.0
  .1","xkey":"XBWN5XQCV3VEKQRH4BVQMJUHWF3OIBSICYW2DAPVTK7UVWUKEL7A3OUX"} ..                                            
##
T 127.0.0.1:9235 -> 127.0.0.1:55290 [AP] #147051
  ..-ERR 'Authentication Timeout'......Authentication Timeout

**Comment by silversword411** [comment]
You're not proxying?
You don't have some in-network security product blocking stuff?

**Reply by optiproplus** [reply]
it's behind nat, in our private cloud. there's no firewall so far, only nat rules

**Reply by silversword411** [reply]
What router manufacturer?

**Reply by optiproplus** [reply]
i'm pretty sure it's not router issue, we have around 100 apps nated inside env without any problem, however if you really wanna dig inm it's netgear prosafe srx53xx model

**Comment by optiproplus** [comment, marked as answer]
i've found the issue, the particular instance were routed via ingress controller which a guess didn't provide all needed headers, after we switched to direct nat it went through flawlessly, thanks for your support
